qcodes/dataset/sqlite_base.py

```diff
@@ -255,8 +256,13 @@ def _convert_numeric(value: bytes) -> Union[float, int, str]:
     # outcome is 'nan'
     if np.isnan(numeric):
         return numeric
+    
+    # If that worked, e.g. did not raise an exception, then we check if the
```

Comment:
I think this comment is misleading (probably just copy-pasted from above). The "If that worked, e.g. did not raise an exception" part does not really make sense here; there is no reason that the nan-check would raise an exception.